src/include/com/amazonaws/kinesis/video/cproducer/Include.h

```diff
@@ -650,6 +650,33 @@ PUBLIC_API STATUS createStaticAuthCallbacks(PClientCallbacks, PCHAR, PCHAR, PCHA
  */
 PUBLIC_API STATUS freeStaticAuthCallbacks(PAuthCallbacks*);
 
+/**
+ * Creates a Credentials auth callbacks from an existing Credential Provider object
+ *
+ * NOTE: The caller is responsible for releasing the structure by calling
+ * the corresponding free API.
+ *
+ * NOTE: The original Credential Provider object will not be released on free
+ *
+ * @param[in] PCallbacksProvider Pointer to callback provider
+ * @param[in] PAwsCredentialProvider Credential provider object
+ * @param[in,out] PAuthCallbacks* Pointer to pointer to AuthCallback struct
+ *
+ * @return STATUS status of operation
+ */
+PUBLIC_API STATUS createCredentialProviderAuthCallbacks(PClientCallbacks, PAwsCredentialProvider, PAuthCallbacks*);
```

Comment:
Do we want to rename this to be more specific for the use case? I am okay with this as well though.

Reply:
The naming reflects the fact that we are supplying a CredentialProvider object which is similar to other overloads